src/Std/Internal/Http/Server.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+module
+
+prelude
+public import Std.Internal.Async
+public import Std.Internal.Async.TCP
+public import Std.Sync.CancellationToken
+public import Std.Internal.Http.Server.Config
+public import Std.Internal.Http.Server.Connection
+
+public section
+
+/-!
+# HTTP Server
+
+This module defines a simple, asynchronous HTTP/1.1 server implementation.
+
+It provides the `Std.Http.Server` structure, which encapsulates all server state, and functions for
+starting, managing, and gracefully shutting down the server.
+
+The server runs entirely using `Async` and uses a shared `CancellationToken` to signal shutdowns.
+Each active client connection is tracked, and the server automatically resolves its shutdown
+promise once all connections have closed.
+-/
+
+namespace Std.Http
+
+open Std.Internal.IO.Async TCP
+
+set_option linter.all true
+
+/--
+The `Server` structure holds all state required to manage the lifecycle of an HTTP server, including
+connection tracking and shutdown coordination.
+-/
+structure Server where
+
+  /--
+  The cancellation token used for shutting down all connections and the server.
+  -/
+  cancellation : Std.CancellationToken
+
+  /--
+  Active HTTP connections
+  -/
+  activeConnections : Std.Mutex UInt64
+
+  /--
+  Indicates when the server has successfully shutdown
+  -/
+  shutdownPromise : IO.Promise Unit
+
+  /--
+  Configuration of the server
+  -/
+  config : Std.Http.Config
+
+namespace Server
+
+/--
+Create a new `Server` structure with an optional configuration.
+-/
+def new (config : Std.Http.Config := {}) : IO Server := do
+  let cancellation ← Std.CancellationToken.new
+  let activeConnections ← Std.Mutex.new 0
+  let shutdownPromise : IO.Promise Unit ← IO.Promise.new
+
+  return { cancellation, activeConnections, shutdownPromise, config }
+
+/--
+Triggers the cancellation of all the requests and the accept loop in the server.
```

Comment:
I think it would be good for the docstrings of all three of `shutdown`, `waitShutdown` and `shutdownAndWait` to contain an explanation of the differences between the three functions and how they are related.